build-logic-settings/cache-plugin/src/main/kotlin/convention-cache.settings.gradle.kts

```diff
@@ -1,18 +1,34 @@
-// See also duplicated settings in parent project
-@Suppress("UnstableApiUsage")
-val avitoGithubRemoteCacheHost: Provider<String> = settings.providers
-    .environmentVariable("GRADLE_CACHE_NODE_HOST")
-    .forUseAtConfigurationTime()
+import com.avito.booleanProperty
+import com.avito.mandatoryStringProperty
 
-val avitoGithubRemoteCachePush: String =
-    extra.properties.getOrDefault("avitoGithub.gradle.buildCache.remote.push", "false").toString()
+// See also duplicated settings in parent project
+val useEnterprise = booleanProperty("avito.gradle.enterprise.enabled", false)
 
 buildCache {
-    remote<HttpBuildCache> {
-        setUrl("http://${avitoGithubRemoteCacheHost.orNull}/cache/")
-        isEnabled = avitoGithubRemoteCacheHost.orNull != null
-        isPush = avitoGithubRemoteCachePush.toBoolean()
-        isAllowUntrustedServer = true
-        isAllowInsecureProtocol = true
+    local {
+        isEnabled = true
+        isPush = true
+        removeUnusedEntriesAfterDays = 30
+    }
+    if (useEnterprise) {
+        remote<HttpBuildCache> {
+            setUrl("${mandatoryStringProperty("avito.gradle.enterprise.url")}/cache/")
+            isEnabled = true
+            isPush = booleanProperty("avito.gradle.buildCache.remote.push", false)
+            isAllowUntrustedServer = true
+            isAllowInsecureProtocol = true
+        }
+    } else {
+        gradle.buildFinished {
```

Comment:
not supported for configuration cache, but acts good: first build prints visible warning (after the build) and following ones will not print anything

looks like a good balance for me without creating ugly build service workarounds

Reply:
> ugly build service workarounds

What are you talking about?

Reply:
> first build prints visible warning (after the build) and following ones will not print anything

My point is no one reads or should read build logs.
I propose to use logs only as side effects and never rely on them in preventing problems.
It's too easy to miss a one-time warning and no one will see the problem.

If we strongly don't want to make this property mandatory we could use tags\values in build scan.

Reply:
> What are you talking about?

https://github.com/runningcode/gradle-doctor/blob/e6cd0299da6e537b05e07ece8ba6813ccccd387b/doctor-plugin/src/main/java/com/osacky/doctor/BuildFinishService.kt

Reply:
> If we strongly don't want to make this property mandatory

New users should set it to false explicitly?

> My point is no one reads or should read build logs.

Bold message in the end on build is hard to miss?

> we could use tags\values in build scan.

it's easy to miss also, but i see no harm adding a tag also

Reply:
> Bold message in the end on build is hard to miss?

The point is about logs in general. I see how to make a solid process around tags in scans or around other events.
With logs only, I have only a hope that a developer will notice it. Hope is not a strategy.

Reply:
I agree with you; let's setup tags after we will make a decision about enterprise

Reply:
>> If we strongly don't want to make this property mandatory

> New users should set it to false explicitly?

Yes, they should. This is inconvenient for new external contributors but more robust than warning in logs.
I don't argue for this solution, just wanted to mention this decision.